cranelift/codegen/src/isa/x64/encoding/evex.rs

```diff
@@ -151,10 +178,44 @@ impl EvexInstruction {
     /// - finally, the ModR/M byte.
     ///
     /// Eventually this method should support encodings of more than just the reg-reg addressing mode (TODO).
-    pub fn encode<CS: ByteSink + ?Sized>(&self, sink: &mut CS) {
+    pub fn encode(&self, sink: &mut MachBuffer<Inst>) {
+        if let RegisterOrAmode::Amode(amode) = &self.rm {
+            if amode.can_trap() {
+                sink.add_trap(TrapCode::HeapOutOfBounds);
+            }
+        }
```

Comment:
It's unfortunate that `sink` has to get more specific... I had always hoped this code could be used for some kind of assembler. But this trap is necessary and if we eventually create an assembler this can get refactored.

Reply:
Oh the other main purpose for a `MachBuffer` is [this use of `use_label_at_offset`](https://github.com/bytecodealliance/wasmtime/blob/5e33c4a3061cf17555eef8b1592a2b653529c55d/cranelift/codegen/src/isa/x64/encoding/rex.rs#L423) for rip-relative addressing modes.